tools/tenscan/frontend/src/components/modules/batches/batch-height-details.tsx

```diff
@@ -0,0 +1,196 @@
+          value={batchDetails?.header?.baseFeePerGas || "-"}
+        />
+        <KeyValueItem
+          label="Inbound Cross Chain Hash"
+          value={
+            <TruncatedAddress
+              address={batchDetails?.header?.inboundCrossChainHash}
+            />
+          }
+        />
+        <KeyValueItem
+          label="Inbound Cross Chain Height"
+          value={Number(batchDetails?.header?.inboundCrossChainHeight)}
+        />
+        <KeyValueItem
+          label="Transfers Tree"
+          value={
+            <TruncatedAddress address={batchDetails?.header?.TransfersTree} />
+          }
+        />
+        <KeyValueItem
+          label="Miner"
+          value={<TruncatedAddress address={batchDetails?.header?.miner} />}
+        />
+        <KeyValueItem
+          label="Base Fee Per Gas"
+          value={formatNumber(batchDetails?.header?.baseFeePerGas)}
+          isLastItem
+        />
+      </KeyValueList>
+      <Separator />
+      <KeyValueList>
+        <KeyValueItem
+          label="No. of Transactions"
+          value={
+            <Link
+              href={`/batch/txs/${batchDetails?.header?.hash}`}
+              className="text-primary"
+            >
+              {batchDetails?.txCount || "-"}
+            </Link>
+          }
+          isLastItem
+        />
+      </KeyValueList>
+      <Separator />
+      <KeyValueList>
+        <KeyValueItem
+          label="Encrypted Tx Blob"
+          value={
+            <>
+              <div className="flex items-center space-x-2">
+                <TruncatedAddress address={batchDetails?.encryptedTxBlob} />{" "}
+                <Button
+                  className="text-sm font-bold leading-none hover:text-primary hover:bg-transparent"
+                  variant="ghost"
+                  onClick={() => {
+                    decryptEncryptedData({
+                      StrData: batchDetails?.encryptedTxBlob,
+                    });
+                    setShowDecryptedData(!showDecryptedData);
+                  }}
+                >
+                  {showDecryptedData && decryptedRollup ? (
+                    <TooltipProvider>
+                      <Tooltip>
+                        <TooltipTrigger>
+                          <EyeClosedIcon className="h-5 w-5 text-muted-foreground hover:text-primary transition-colors cursor-pointer ml-2" />
+                        </TooltipTrigger>
+                        <TooltipContent>Hide Encrypted Data</TooltipContent>
+                      </Tooltip>
+                    </TooltipProvider>
+                  ) : (
+                    <TooltipProvider>
+                      <Tooltip>
+                        <TooltipTrigger>
+                          <EyeOpenIcon className="h-5 w-5 text-muted-foreground hover:text-primary transition-colors cursor-pointer ml-2" />
+                        </TooltipTrigger>
+                        <TooltipContent>Show Encrypted Data</TooltipContent>
```

Comment:
I think this layout needs a bit of tweaking - it looks a little messy on the page. I think we should probs remove the copy encrypted blob icon since that is only for pasting it into the decryption page that we don't need anymore.

How would it look to remove the "Show/ Hide" labels since the view button is sort of self explanatory.

Reply:
The text is just a tooltip that shows when you hover over the icon. It’s not always there.